src/platform/forms-system/src/js/containers/FormPage.jsx

```diff
@@ -45,6 +47,7 @@ class FormPage extends React.Component {
         this.props.route.pageConfig.pageKey ||
       get('params.index', prevProps) !== get('params.index', this.props)
```

Comment:
Triggered once per every page change. Every form.

This seems appropriate place to put this to me, but open to trying other ideas too. 

Although this check will be on every form, it is immediately only checking what is necessary before it will fail or proceed - pageConfig.showPagePerItem && pageConfig.allowPathWithNoItems - if yes, then we are going to prepopulate some formData for you for this item (with empty values) for the given path, e.g.   array-path/0, array-path/1, etc.  

this solves a lot of issues with RJSF about determining what needs to be required, what updateSchemas will be called, hidden fields, and a lot of downstream functions that will automatically be taken care of by adding this.